Source/WebCore/layout/formattingContexts/inline/InlineLine.cpp

```diff
@@ -923,7 +923,7 @@ InlineLayoutUnit Line::Run::removeTrailingWhitespace()
         auto& inlineTextBox = downcast<InlineTextBox>(*m_layoutBox);
         auto startPosition = *m_lastNonWhitespaceContentStart;
         auto endPosition = m_textContent->start + m_textContent->length;
-        RELEASE_ASSERT(startPosition < endPosition - trailingTrimmableContentLength);
+        RELEASE_ASSERT(startPosition <= endPosition - trailingTrimmableContentLength);
```

Comment:
This assert indicates that there's a gap in the line run. I did a bit of debugging and it looks like we incorrectly "expand" (see Line::Run::expand) a run after a skipped whitespace (see Line::appendTextContent -> willCollapseCompletely) -and we can only expand on continuous content (without gaps in between inline text items). It's very curious.

Reply:
It looks like Line::appendTextContent receives 2 whitespace type of InlineTextItems in a row. This is unexpected and confuses the run expand logic (normally we don't create separate InlineItems for adjacent whitespace characters).

Reply:
```
<style>
body { 
  white-space: pre-line;
  -webkit-text-combine: horizontal;
  -webkit-mask-image: url(data:image/gif;base64,iVB);
}

*:focus-within {
  text-wrap: nowrap;
}
</style>
<script>
function main() {
  document.styleSheets[0].media.mediaText = "print (min-height:10px)";
  document.dir = "rtl";
}
</script>
<body onload="main()">
<span></span>
AAA
 A AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA<textarea autofocus="">

```
here is a slightly simplified version of the test case.